### ISSUE TITLE
onKeyDown swallows events

### ISSUE THREAD

**Issue by jeffreysilver** [jasonphillips/slate-deep-table#6]
The onKeyDown implementation stops event propagation, leading to some whacky behavior as we integrate this plugin to Slate.

One way to prevent this is by calling `next()` when the current selection is not in the table, letting other `onKeyDown` implementations handle the event. I'm happy to create a pull request that does the following:

```
    /**
     * User is pressing a key in the editor
     */
    function onKeyDown(event, change, next) {
        const { value } = change;
        // Only handle events in cells
        if (!isSelectionInTable(value)) {
            return next();
        }

        // Build arguments list
        const args = [ event, change, opts ];

        switch (event.key) {
            case KEY_TAB:
                return onTab(...args);
            case KEY_DOWN:
            case KEY_UP:
                return onUpDown(...args);
        }
    }
```

**Comment by jasonphillips** [comment]
Good call; yes, this is due to a breaking change as of **Slate 0.42**, as documented here: 

https://github.com/ianstormtaylor/slate/blob/master/packages/slate/Changelog.md#0420--october-9-2018

> The middleware stack must now be explicitly continued, using next. Previously returning undefined from a middleware would (usually) continue the stack onto the next middleware. Now, with middleware taking a next function argument you must explicitly decide to continue the stack by call next() yourself.

The latest version of Slate for which I've adapted this library is **0.41**.  I usually just indicate the compatibility by way of the peer dependencies: 
https://github.com/jasonphillips/slate-deep-table/blob/master/package.json#L15

--but given the speed of breaking Slate changes lately, I think it's worth putting a small guide or note at the top of the README soon, to list which precise version of this library to grab for which Slate iteration you're currently on.

In any case, adapting to 0.42 (and then 0.43, which released in just the past few days with additional breaking changes!) is certainly needed. I had waited a bit because I noticed a couple of forks of this repo out there which have already updated for 0.42 (including the next() keyhandler update mentioned here, it appears), and hoped one of them would become a PR:

https://github.com/Yevs/slate-deep-table
https://github.com/knowledge-point/slate-deep-table

@Yevs or @knowledge-point -- any interest in submitting your work to a PR here?

**Comment by jasonphillips** [comment]
Closed by #8 - check the [changelog](https://github.com/jasonphillips/slate-deep-table/blob/master/CHANGELOG.md) and updated [example](https://github.com/jasonphillips/slate-deep-table/blob/master/example/main.js) for some updates to syntax and breaking changes.
